src/utils/openvpn.ts

```diff
@@ -267,26 +277,66 @@ export class VpnManager extends EventEmitter {
 		return emitter;
 	};
 
-	public start(): Bluebird<true> {
-		this.process = spawn('/usr/sbin/openvpn', this.args(), {
-			stdio: ['ignore', 'pipe', 'pipe'],
-		});
-		// proxy logs from the child process stdout/stderr
-		this.process.stdout.pipe(es.split()).on('data', data => {
-			this.emit('log', VpnLogLevels.n, data);
-		});
-		this.process.stderr.pipe(es.split()).on('data', data => {
-			this.emit('log', VpnLogLevels.n, data);
-		});
-		// proxy error events from the child process
-		this.process.on('error', err => {
-			this.emit('process:error', err);
-		});
-		this.process.on('exit', (code, signal) => {
-			this.emit('process:exit', code, signal);
+	private async killOrphan() {
+		try {
+			await fs.promises.access(this.pidFile);
+		} catch {
+			return;
+		}
+
+		const pid = (await fs.promises.readFile(this.pidFile, 'utf8')).replace(
+			/\D/g,
+			'',
+		);
+		const signals = { PING: 0, KILL: 9 } as const;
+		const kill = (signal?: ValueOf<typeof signals>) =>
+			spawn('/bin/kill', signal != null ? [`-${signal}`, pid] : [pid]);
+
+		return new Promise((resolve, reject) => {
+			const start = Date.now();
+			const waitForDeath = (code?: number) => {
+				if (code !== 0) {
+					// the signal was unsuccessful, the pid no longer exists;
+					// the process is dead, and our work here is done.
+					return resolve();
+				}
+				const now = Date.now();
```

Comment:
You could make this
```suggestion
				const elapsedTime = Date.now() - start;
```
and then use
```ts
if (elapsedTime  > TERM_TIMEOUT) {
```
which is a little more readable imo

Reply:
Agreed, will change to this. Cheers.